src/components/Divider.tsx

```diff
@@ -1,70 +1,141 @@
 import { SystemStyleObject } from '../../styled-system/types'
 import Path from '../@types/Path'
+import State from '../@types/State'
+import ThoughtId from '../@types/ThoughtId'
 import { setCursorActionCreator as setCursor } from '../actions/setCursor'
-import { DIVIDER_MIN_WIDTH, DIVIDER_PLUS_PX } from '../constants'
+import { DIVIDER_MIN_WIDTH } from '../constants'
 import attributeEquals from '../selectors/attributeEquals'
+import { getAllChildrenAsThoughts } from '../selectors/getChildren'
 import rootedParentOf from '../selectors/rootedParentOf'
+import editingValueUntrimmedStore from '../stores/editingValueUntrimmed'
 import fastClick from '../util/fastClick'
 import head from '../util/head'
+import isDivider from '../util/isDivider'
+
+/** Custom hook to fetch divider-related data from the state. */
+const useDividerData = (path: Path) => {
+  return useSelector(
+    (state: State) => {
+      const parentPath = rootedParentOf(state, path)
+      const parentId = head(parentPath)
+      const grandParentPath = parentId ? rootedParentOf(state, parentPath) : []
+      const grandParentId = head(grandParentPath)
+      const children = parentId ? getAllChildrenAsThoughts(state, parentId) : []
+      const childrenWithoutDividers = children.filter(child => !isDivider(child.value))
+      const isOnlyChild = childrenWithoutDividers.length === 0
+      const isTableView =
+        (parentId && attributeEquals(state, parentId, '=view', 'Table')) ||
+        (grandParentId && attributeEquals(state, grandParentId, '=view', 'Table'))
+
+      let thoughtsAtSameDepth: { id: ThoughtId; value: string }[] = []
+
+      if (isTableView && isOnlyChild && grandParentId) {
+        const parentSiblings = getAllChildrenAsThoughts(state, grandParentId).filter(child => !isDivider(child.value))
+        thoughtsAtSameDepth = parentSiblings.flatMap(parent => {
+          const childrenOfParent = parent.id ? getAllChildrenAsThoughts(state, parent.id) : []
+          return childrenOfParent.filter(child => !isDivider(child.value))
+        })
+      }
+
+      return {
+        isOnlyChild,
+        isTableView,
+        children,
+        thoughtsAtSameDepth,
+      }
+    },
+    // Ensure the selector runs only when values change, excluding ref changes to avoid redundant re-renders.
+    (prev, next) =>
+      prev.isOnlyChild === next.isOnlyChild &&
+      prev.isTableView === next.isTableView &&
+      prev.children.length === next.children.length &&
+      prev.children.every((child, index) => child.value === next.children[index].value) &&
+      prev.thoughtsAtSameDepth.length === next.thoughtsAtSameDepth.length &&
+      prev.thoughtsAtSameDepth.every((thought, index) => thought.value === next.thoughtsAtSameDepth[index].value),
+  )
+}
+
+/**
+ * Calculates the width of multiple thoughts by measuring their rendered widths in the DOM.
+ */
+const getThoughtWidths = (thoughts: { id: ThoughtId }[]): number[] => {
```

Comment:
Since this function only needs ids, it would be better to reduce the surface area of the parameters and just take `ids: ThoughtId[]`.